### PR TITLE
contributors/devel/sig-instrumentation: Add further details on naming best practices

### PR DESCRIPTION
**Pull request by brancz**
Follow up of https://github.com/kubernetes/kubernetes/pull/72470#issuecomment-529007782

cc @logicalhan @piosz @mikedanese @ehashman @LiliC @s-urbaniak

### CODE REVIEW

File: contributors/devel/sig-instrumentation/instrumentation.md [hunk 1]
```diff
@@ -90,7 +90,8 @@ apply additionally.
 
 ## Naming 
 
-Metrics added directly by application or package code should have a unique name. 
+General [metric and label naming apply](https://prometheus.io/docs/practices/naming/),
```
Comment (s-urbaniak) [inline contributors/devel/sig-instrumentation/instrumentation.md]: `metric and label naming best practices apply`?

Comment (brancz) [reply contributors/devel/sig-instrumentation/instrumentation.md]: oops yes

File: contributors/devel/sig-instrumentation/instrumentation.md [hunk 2]
```diff
@@ -90,7 +90,8 @@ apply additionally.
 
 ## Naming 
 
-Metrics added directly by application or package code should have a unique name. 
+General [metric and label naming best practices](https://prometheus.io/docs/practices/naming/) apply.
+Beyond that, metrics added directly by application or package code should have a unique name.
```
[the remark this answers is not in the document]
Comment (brancz) [reply contributors/devel/sig-instrumentation/instrumentation.md]: That's a non issue, as the Prometheus registry will panic when a metric with the same name is registered twice.

**Comment by mikedanese** [comment]
Thanks.

lgtm without auto submit so others get a chance to review

File: contributors/devel/sig-instrumentation/instrumentation.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (logicalhan) [reply contributors/devel/sig-instrumentation/instrumentation.md]: Only if you use `MustRegister`, calls to `Register` can fail silently.

**Comment by mikedanese** [comment]
I guess I forgot to hit refresh.

/shrug

**Pull request merged**